src/app/core/common-components/entity-select/entity-select.component.ts

```diff
@@ -89,8 +94,10 @@ export class EntitySelectComponent<E extends Entity> implements OnChanges {
         filter((isLoading) => !isLoading),
       )
       .subscribe((_) => {
-        this.selectedEntities = this.allEntities.filter((e) =>
-          sel.find((s) => s === e.getId(true) || s === e.getId()),
+        this.selectedEntities = sel.map((id) =>
+          this.allEntities.find(
+            (s) => id === s.getId(true) || id === s.getId(),
+          ),
```

Comment:
what's supposed to happen here if an item in "sel" is not available in allEntities? Should this really put an undefined value in the selectedEntities array?